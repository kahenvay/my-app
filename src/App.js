import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Link} from 'react-router-dom';
import { browserHistory } from 'react-router';
import PageHeader from './components/layout/PageHeader';
import PageFooter from './components/layout/PageFooter';
import HomePage from './pages/HomePage';
import logo from './logo.svg';
import './App.css';



class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <PageHeader />
          <main><Route name="home" exact path="/" component={HomePage} /></main>
          <PageFooter />
        </div>
      </Router>
    );
  }
}

export default App;
