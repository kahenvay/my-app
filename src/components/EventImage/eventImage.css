.event__image img{    
    object-fit: cover;
    width: 100%;
    height: 100%;

}