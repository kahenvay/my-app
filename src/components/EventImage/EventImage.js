import React from 'react';
import PropTypes from 'prop-types';
import './eventImage.css';

const EventImage = ({image, title}) =>{

    return(
        <div className="event__image">
            <img src={image} alt={title} />
        </div>
    )
}

EventImage.propTypes = {
    image : PropTypes.string,
    title: PropTypes.string
};

export default EventImage;