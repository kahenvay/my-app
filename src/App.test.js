import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { configure } from 'enzyme';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json";
import Adapter from 'enzyme-adapter-react-16';
import EventImage from './components/EventImage/EventImage';
configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

/*it("should render the homepage and the events", () => {
  const props = {
    events: [],
    isLoading: true,
    errors: null
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()
})*/

it('render correctly EventImage component', () => {  
  const EventImageComponent = renderer.create(<EventImage />).toJSON();
  expect(EventImageComponent).toMatchSnapshot();
});