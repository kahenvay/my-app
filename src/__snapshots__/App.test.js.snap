// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the homepage and the events 1`] = `
<App
  errors={null}
  isLoading={true}
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <PageHeader>
          <header>
            <div
              className="header__top"
            >
              <div
                className="logo_container"
              >
                <img
                  alt="Gradient banner"
                  className="logo"
                  src="https://z-m-scontent.flis5-1.fna.fbcdn.net/v/t1.0-9/50481853_2500649853342915_5506782981307498496_n.jpg?_nc_cat=108&_nc_ad=z-m&_nc_cid=1272&_nc_zor=9&_nc_ht=z-m-scontent.flis5-1.fna&oh=7c00ad8687755a389dcec549346b4de1&oe=5CFAB789"
                />
              </div>
              <NavBar>
                <ul
                  className="headerButtons"
                >
                  <li
                    className="navButton"
                  >
                    <Link
                      replace={false}
                      to=""
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </Link>
                  </li>
                </ul>
              </NavBar>
            </div>
            <img
              alt="Gradient banner from ucf.org.au"
              className="header__banner"
              src="http://ucf.org.au/wp-content/uploads/ucf-gradient-banner.jpg"
            />
          </header>
        </PageHeader>
        <main>
          <Route
            component={[Function]}
            exact={true}
            name="home"
            path="/"
          >
            <HomePage
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div>
                <h1>
                  MIX Movement
                </h1>
                <h2>
                  Latest events
                </h2>
                <div>
                  <Message
                    errors={null}
                  >
                    <div>
                      <p>
                        Loading
                      </p>
                    </div>
                  </Message>
                </div>
              </div>
            </HomePage>
          </Route>
          <Route
            component={[Function]}
            exact={true}
            name="single event"
            path="/event/:id"
          />
        </main>
        <Footer>
          <footer>
            <div>
              <p>
                Footer component
              </p>
            </div>
          </footer>
        </Footer>
      </div>
    </Router>
  </BrowserRouter>
</App>
`;
